test/gtest/any.hpp

```diff
@@ -0,0 +1,223 @@
+    {
+    }
+};
+
+inline std::vector<AnyTestCase> AnyTestConfigs()
+{
+    return {
+        // TODO: Handle cases where input params has zero dim(s)
+        // AnyTestCase({3, 0, 4, 5}),
+        AnyTestCase({3, 4, 5}, -1, false),
+        AnyTestCase({3, 4, 5}, -1, false, false),
+        AnyTestCase({4, 5, 7, 8}),
+        AnyTestCase({4, 5, 7, 8}, -1, false, false),
+        AnyTestCase({4, 5, 7, 8}, 0),
+        AnyTestCase({4, 5, 7, 8}, 0, true),
+        AnyTestCase({5}),
+        AnyTestCase({4, 5}),
+        AnyTestCase({4, 5, 7}),
+        AnyTestCase({4, 5, 7}, 0),
+        AnyTestCase({4, 5, 7}, 0, true),
+        AnyTestCase({4, 5, 7}, 1),
+        AnyTestCase({4, 5, 7}, 1, true),
+        AnyTestCase({4, 5, 7}, 2),
+        AnyTestCase({4, 5, 7}, 2, true),
+        AnyTestCase({4, 5, 7, 8}, 3),
+        AnyTestCase({4, 5, 7, 8}, 3, true),
+    };
+}
+
+template <typename T>
+struct AnyTest : public ::testing::TestWithParam<AnyTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle = get_handle();
+        any_config    = GetParam();
+
+        auto in_dims = any_config.GetInputShape();
+        dim          = any_config.dim;
+        keepdim      = any_config.keepdim;
+
+        auto gen_in_value = [](auto...) {
+            return prng::gen_A_to_B<T>(std::numeric_limits<T>::min(),
+                                       std::numeric_limits<T>::max());
+        };
+
+        if(any_config.is_contiguous)
+        {
+            input = tensor<T>{in_dims}.generate(gen_in_value);
+        }
+        else
+        {
+            std::vector<size_t> in_strides(in_dims.size());
+            in_strides.back() = 1;
+            for(int i = in_dims.size() - 2; i >= 0; --i)
+                in_strides[i] = in_strides[i + 1] * in_dims[i + 1];
+            in_strides[0] *= 2;
+            input = tensor<T>{in_dims, in_strides}.generate(gen_in_value);
+        }
+
+        input_dev = handle.Write(input.data);
+
+        std::vector<size_t> out_dims(in_dims);
+        if(dim != -1)
+        {
+            if(keepdim)
+            {
+                out_dims[dim] = 1;
+            }
+            else
+            {
+                out_dims.erase(out_dims.begin() + dim);
+            }
+        }
+        else
+        {
+            out_dims = {1};
+        }
+
+        output = tensor<unsigned char>{out_dims};
+        std::fill(output.begin(), output.end(), 0);
+
+        ref_output = tensor<unsigned char>{out_dims};
+        std::fill(ref_output.begin(), ref_output.end(), 0);
+
+        output_dev = handle.Write(output.data);
+
+        ws_sizeInBytes =
+            miopen::GetAnyForwardWorkspaceSize(handle, input.desc, output.desc, dim, keepdim);
+
+        if(ws_sizeInBytes == static_cast<size_t>(-1))
+            GTEST_FAIL() << "Call GetAnyForwardWorkspaceSize failed!";
+
+        if(ws_sizeInBytes > 0)
+        {
+            workspace = tensor<float>{ws_sizeInBytes / sizeof(float)};
```

Comment:
May I asked if workspace is tensor<float> or tensor<T>, because this code and code in `forward_any.cpp` to get workspace size seems to not match with each other

Reply:
it was tensor<T>. I fixed this issue.